components/br-stream/src/metadata/keys.rs

```diff
@@ -0,0 +1,148 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+use bytes::BufMut;
+use lazy_static::lazy_static;
+use regex::Regex;
+
+const PREFIX: &str = "/tidb/br-stream";
+const PATH_INFO: &str = "/info";
+const PATH_NEXT_BACKUP_TS: &str = "/checkpoint";
+const PATH_RANGES: &str = "/ranges";
+const PATH_PAUSE: &str = "/pause";
+lazy_static! {
+    static ref EXTRACT_NAME_FROM_INFO_RE: Regex =
+        Regex::new(r"/tidb/br-stream/info/(?P<task_name>[0-9a-zA-Z_]+)").unwrap();
+}
+
+/// A key that associates to some metadata.
+///
+/// Generally, there are four pattern of metadata:
+/// ```plaintext
+/// For basic task info:
+/// <PREFIX>/info/<task_name> -> <task_info(protobuf)>
+/// For the target ranges of some task:
+/// <PREFIX>/ranges/<task_name>/<start_key(binary)> -> <end_key(binary)>
+/// For the progress of tasks:
+/// <PREFIX>/checkpoint/<task_name>/<store_id(u64,be)>/<region_id(u64,be)> -> <next_backup_ts(u64,be)>
+/// For the status of tasks:
+/// <PREFIX>/pause/<task_name> -> ""
+/// ```
+#[derive(Clone)]
+pub struct MetaKey(pub Vec<u8>);
+
+/// A simple key value pair of metadata.
+#[derive(Clone, Debug)]
+pub struct KeyValue(pub MetaKey, pub Vec<u8>);
+
+impl std::fmt::Debug for MetaKey {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let s = match std::str::from_utf8(&self.0) {
+            Ok(s) => s.to_owned(),
+            Err(_) => format!("<{}>", hex::encode(&self.0)),
+        };
+        f.debug_tuple("MetaKey").field(&s).finish()
```

Comment:
https://doc.rust-lang.org/std/primitive.slice.html#method.escape_ascii

```suggestion
        f.debug_tuple("MetaKey").field(&self.0.escape_ascii()).finish()
```